api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java

```diff
@@ -137,11 +136,17 @@ public void createMetric(
     })
     public void getMetrics(
             @Suspended AsyncResponse asyncResponse,
-            @ApiParam(value = "List of tags filters", required = false) @QueryParam("tags") Tags tags) {
+            @ApiParam(value = "List of tags filters", required = false) @QueryParam("tags") Tags tags,
+            @ApiParam(value = "Tags query expression", required = false) @QueryParam("tagsQuery") String tagsQuery) {
 
-        Observable<Metric<AvailabilityType>> metricObservable = (tags == null)
-                ? metricsService.findMetrics(getTenant(), AVAILABILITY)
-                : metricsService.findMetricsWithFilters(getTenant(), AVAILABILITY, tags.getTags());
+        Observable<Metric<AvailabilityType>> metricObservable = null;
+        if (tags != null && tagsQuery == null) {
```

Comment:
Is it really required to support both?  I guess for backward compatibility?  Perhaps 'tags' should support both formats.